### PR TITLE
Add setting to hide netcore installation prompt

### PR DESCRIPTION
**Pull request by Benjin**
Fixes https://github.com/microsoft/azuredatastudio/issues/15247 by adding a "Don't ask again" option to not prompt the user to install .NET core if it's not found.

However, adding a third button caused everything to get truncated:
image

so I had to update the strings a bit:
image

### CODE REVIEW

**Comment by Charles-Gagnon** [comment]
What about `Update location` `Install` and `Don't ask again` as the options?

**Comment by Benjin** [comment]
Yup, that's pretty much what I was thinking.  Any additional definition can be (/is already) done in the text above.

**Review by Charles-Gagnon — APPROVED** [review]

File: extensions/sql-database-projects/src/tools/netcoreTool.ts [hunk 1]
```diff
@@ -35,23 +37,26 @@ export class NetCoreTool {
...
+		if (!this.isNetCoreInstallationPresent && vscode.workspace.getConfiguration(DBProjectConfigurationKey)[NetCoreDoNotAskAgainKey] !== true) {
 			await this.showInstallDialog();
 			return false;
 		}
 		return true;
 	}
 
 	public async showInstallDialog(): Promise<void> {
-		let result = await vscode.window.showInformationMessage(NetCoreInstallationConfirmation, UpdateNetCoreLocation, InstallNetCore);
+		let result = await vscode.window.showInformationMessage(NetCoreInstallationConfirmation, UpdateNetCoreLocation, InstallNetCore, DoNotAskAgain);
```
Comment (Benjin) [inline extensions/sql-database-projects/src/tools/netcoreTool.ts:48]: Something about this ordering feels a bit weird now that "Don't Ask" is included... @radentonev, is there a preferred order for change/positive/ignore buttons?

Comment (Benjin) [reply extensions/sql-database-projects/src/tools/netcoreTool.ts:48]: This article was an neat read: https://uxplanet.org/primary-secondary-action-buttons-c16df9b36150

File: extensions/sql-database-projects/src/tools/netcoreTool.ts [hunk 2]
```diff
@@ -15,10 +15,12 @@ const localize = nls.loadMessageBundle();
...
 export const DBProjectConfigurationKey: string = 'sqlDatabaseProjects';
 export const NetCoreInstallLocationKey: string = 'netCoreSDKLocation';
+export const NetCoreDoNotAskAgainKey: string = 'netCoreDoNotAsk';
 export const NextCoreNonWindowsDefaultPath = '/usr/local/share';
 export const NetCoreInstallationConfirmation: string = localize('sqlDatabaseProjects.NetCoreInstallationConfirmation', "The .NET Core SDK cannot be located. Project build will not work. Please install .NET Core SDK version 3.1 or update the .Net Core SDK location in settings if already installed.");
-export const UpdateNetCoreLocation: string = localize('sqlDatabaseProjects.UpdateNetCoreLocation', "Update .Net Core location");
-export const InstallNetCore: string = localize('sqlDatabaseProjects.InstallNetCore', "Install .Net Core SDK");
+export const UpdateNetCoreLocation: string = localize('sqlDatabaseProjects.UpdateNetCoreLocation', "Update Location");
+export const InstallNetCore: string = localize('sqlDatabaseProjects.InstallNetCore', "Install");
+export const DoNotAskAgain: string = localize('sqlDatabaseProjects.doNotAskAgain', "Don't Ask Again");
```
Comment (kisantia) [inline extensions/sql-database-projects/src/tools/netcoreTool.ts:23]: can you also update the DoNotShowAgain string in data-workspace to also be "Don't Ask Again" for consistency?

**Review by kisantia — APPROVED** [review]

**Comment by kburtram** [comment]
I think a lot of times the "Don't Show Again" is a menu item that is available from the Gear icon.  You could consider that if the buttons are too crowded.

File: extensions/sql-database-projects/src/tools/netcoreTool.ts (hunk 1 above)
Comment (udeeshagautam) [reply extensions/sql-database-projects/src/tools/netcoreTool.ts:48]: I like Karl's idea too. For real customers, not tests or internal folks testing, I do think the install options are more important to show up front so their builds dont fail...

Comment (Benjin) [reply extensions/sql-database-projects/src/tools/netcoreTool.ts:48]: Looks like that option isn't exposed to extensions; primary actions only.

**Pull request merged**